napi_basics.md

```diff
@@ -0,0 +1,120 @@
+# N-API - Basics
+
+N-API is an API exposed by Node to developers that allow them to author Native add-ons 
+in an ABI stable manner. This means that these add-ons can work against future versions
+of Node without recompilation, and developers get the additional benefit that their 
+add-ons can work independent of the underlying JavaScript Virtual Machine being used 
+by Node.
+
+APIs exposed in the N-API surface are generally used to create and manipulate JavaScript
+objects. Concepts and operations generally map to ideas specified in the ECMA262 Language
+SPecification. The APIs generally have the following properties:
+- All N-API APIs generally return a status code of type `napi_status`. This status 
+can be used to determine whether the API call succeeded or failed.
+- The actual value being returned by the API is returned in an out parameter.
+- All JavaScript objects are abstracted behind an opaque type named `napi_value`.
+- In case of an error status code, additional information can be obtained using 
+`napi_get_last_error_info`
+
+## N-API types
+
+N-API exposes the following fundamental datatypes as abstractions that are consumed by 
+the various APIs. These APIs should be treated as opaque types that can be introspected
+only with other N-API APIs.
+
+### *napi_env*
+
+`napi_env` is used to represent a context that the underlying N-API implementation can use 
+to persist VM specific state.
+<< REVIEW: What else can we say here? >>
+
+### *napi_value*
+This is an opaque pointer that is used to represent a JavaScript object.
+
+### *napi_ref*
+This is the abstraction to use to reference a `napi_value`. This allows for users 
+to manage the lifetimes of JavaScript objects, including defining their minimum
+lifetimes explicitly, and also being notified when they are garbage-collected.
```

Comment:
Talked to @jasongin offline- the part I was missing was that napi_wrap cannot be called on arbitrary objects so there is no current mechanism in NAPI to be notified when an arbitrary napi_value is collected. Updating the doc.